cpp/src/bitmask/null_mask.cu

```diff
@@ -261,14 +264,15 @@ __global__ void count_set_bits_kernel(bitmask_type const* bitmask,
 
   auto const first_word_index{word_index(first_bit_index)};
   auto const last_word_index{word_index(last_bit_index)};
-  auto const tid         = threadIdx.x + blockIdx.x * blockDim.x;
-  auto thread_word_index = tid + first_word_index;
+  thread_index_type const tid         = threadIdx.x + blockIdx.x * blockDim.x;
+  thread_index_type const stride      = blockDim.x * gridDim.x;
+  thread_index_type thread_word_index = tid + first_word_index;
```

Comment:
Probably we also need to upgrade `first_` and `last_` indices above.

Reply:
We can reply on implicit conversions for `<` or `<=` since they are always cast to the larger data type for comparison. `first_word_index` and `last_word_index` are explicitly used as `size_type` later in the loop thus setting them to `thread_index_type` seems not worth the effort.

Reply:
I think I understand, but I'd better confirm:
1. `first_word_index` and `last_word_index` are `int32_t`.
2. `tid` is `uint32_t`. So `thread_word_index` is eventually `uint32_t`.
3. `while (thread_word_idx <= last_word_index)` compares the two as `uint32_t`.
4. The place where `thread_word_index` might overflow is at line 275 (`thread_word_index += stride`). But both `thread_word_index` and `stride` are `uint32_t` already.